### PR TITLE
Make `ChainMap`'s `new_child()` and `parents` return subclass

### PR DESCRIPTION
**Pull request by jack1142**
This works incorrectly both for subclasses of Python 3.8's `typing.ChainMap` and Python 3.9's `collections.ChainMap`, I'm not sure if this is enough to fix it for `typing.ChainMap` and would like to know what I can do to fix it for that as well if this is not enough.

### CODE REVIEW

File: stdlib/collections/__init__.pyi [hunk 1]
```diff
@@ -286,9 +286,9 @@ class defaultdict(Dict[_KT, _VT], Generic[_KT, _VT]):
 class ChainMap(MutableMapping[_KT, _VT], Generic[_KT, _VT]):
     maps: list[Mapping[_KT, _VT]]
     def __init__(self, *maps: Mapping[_KT, _VT]) -> None: ...
-    def new_child(self, m: Mapping[_KT, _VT] | None = ...) -> ChainMap[_KT, _VT]: ...
+    def new_child(self: _S, m: Mapping[_KT, _VT] | None = ...) -> _S: ...
     @property
-    def parents(self) -> ChainMap[_KT, _VT]: ...
+    def parents(self: _S) -> _S: ...
```
Comment (srittau) [inline stdlib/collections/__init__.pyi:291]: Nit: There's a `_typeshed.Self` type alias, which we prefer for those self types (as a marker).

**Review by srittau — CHANGES_REQUESTED** [review]

File: stdlib/collections/__init__.pyi (hunk 1 above)
Comment (jack1142) [reply stdlib/collections/__init__.pyi]: Fixed, I originally based it off defaultdict's `copy()` which I guess was missed when `_typeshed.Self` was introduced.

**Comment by jack1142** [comment]
pytype run failed but with no details, should I just close&reopen to rerun it?

**Comment by srittau** [comment]
I restarted CI.

**Review by srittau — APPROVED** [review]

**Pull request merged**